File: packages/dashboard/src/types/trading-account.ts
export type BrokerProvider = 'metaapi' | 'deriv' | 'alpaca' | 'binance' | 'ibkr' | 'stub';
export type AccountKind = 'personal' | 'prop' | 'demo';

export interface TradingAccount {
  id: string;
  userId: string;
  metaapiAccountId: string | null;
  label?: string;
  isActive: boolean;
  brokerProvider?: BrokerProvider | null;
  accountKind?: AccountKind;
  // MetaAPI / MT5
  mt5Login?: string;
  mt5Server?: string;
  // Deriv (token is never sent to the client)
  derivLoginId?: string | null;
  createdAt: string;
}

export interface AccountDetails {
  balance: number;
  equity: number;
  margin: number;
  free_margin: number;
  open_positions: number;
  leverage: number;
  state?: string;           // MetaAPI state: DEPLOYED, UNDEPLOYED, DEPLOYING, etc.
  connection_status?: string; // CONNECTED, DISCONNECTED, TIMEOUT, etc.
}

export interface CreateAccountDto {
  brokerProvider?: 'metaapi' | 'deriv';
  accountKind?: AccountKind;
  label?: string;
  // MetaAPI / MT5 flow
  login?: string;
  password?: string;
  serverName?: string;
  platform?: 'mt5' | 'mt4';
  // Deriv-direct flow
  derivApiToken?: string;
  derivLoginId?: string;
}

File: packages/execution-engine/src/models/trading_account.py
from typing import Optional

from pydantic import BaseModel


class TradingAccount(BaseModel):
    id: str  # UUID
    user_id: str  # UUID
    metaapi_account_id: Optional[str] = None  # MetaAPI account ID (None for Deriv-direct accounts)
    label: Optional[str] = None
    is_active: bool = True

    # --- Routing & funding ---
    account_kind: str = "personal"          # 'personal' | 'prop' | 'demo'
    broker_provider: Optional[str] = None   # Explicit override: 'deriv'|'metaapi'|... or None to auto-route

    # --- Deriv-direct credentials (only set for brokerProvider='deriv') ---
    deriv_api_token: Optional[str] = None
    deriv_login_id: Optional[str] = None

    # --- Prop firm settings (null on personal/demo) ---
    prop_firm_name: Optional[str] = None
    prop_max_daily_loss_pct: Optional[float] = None      # e.g. 5.0 for 5%
    prop_max_total_drawdown_pct: Optional[float] = None  # e.g. 10.0 for 10%
    prop_profit_target_pct: Optional[float] = None       # e.g. 8.0 for 8%

    # --- Runtime state (not persisted) ---
    equity: float = 0.0
    balance: float = 0.0
    open_positions: int = 0
    daily_loss: float = 0.0
    total_lot_exposure: float = 0.0
